modules/directives/src/main/scala/scala/build/preprocessing/directives/Toolkit.scala

```diff
@@ -32,32 +35,70 @@ final case class Toolkit(
   testToolkit: Option[Positioned[String]] = None
 ) extends HasBuildOptionsWithRequirements {
   def buildOptionsList: List[Either[BuildException, WithBuildRequirements[BuildOptions]]] =
-    toolkit.map(t => Toolkit.buildOptions(t).map(_.withEmptyRequirements)).toList ++
-      testToolkit.map(tt => Toolkit.buildOptions(tt).map(_.withScopeRequirement(Scope.Test))).toList
+    Toolkit.buildOptionsWithScopeRequirement(toolkit, defaultScope = None) ++
+      Toolkit.buildOptionsWithScopeRequirement(testToolkit, defaultScope = Some(Scope.Test))
 }
 
 object Toolkit {
-  def resolveDependency(toolkitCoords: Positioned[String])
-    : Positioned[DependencyLike[NameAttributes, NameAttributes]] =
-    toolkitCoords.map(coords =>
-      val tokens  = coords.split(':')
-      val version = tokens.last
-      val v       = if version == "latest" then "latest.release" else version
-      val flavor  = tokens.dropRight(1).headOption
-      val org = flavor match {
-        case Some("typelevel") => Constants.typelevelOrganization
-        case Some(org)         => org
-        case None              => Constants.toolkitOrganization
-      }
-      dep"$org::${Constants.toolkitName}::$v,toolkit"
-    )
+  def resolveDependenciesWithRequirements(toolkitCoords: Positioned[String])
+    : List[WithBuildRequirements[Positioned[DependencyLike[NameAttributes, NameAttributes]]]] =
+    toolkitCoords match
+      case Positioned(positions, coords) =>
+        val tokens  = coords.split(':')
+        val version = tokens.last
+        val v       = if version == "latest" then "latest.release" else version
+        val flavor  = tokens.dropRight(1).headOption
+        val org = flavor match {
+          case Some("typelevel") => Constants.typelevelOrganization
+          case Some(org)         => org
+          case None              => Constants.toolkitOrganization
+        }
+        if org == Constants.toolkitOrganization && (version == "latest" || v.asVersion > "0.1.6".asVersion)
+        then
+          List(
+            Positioned(positions, dep"$org::${Constants.toolkitName}::$v,toolkit")
+              .withEmptyRequirements,
+            Positioned(positions, dep"$org::${Constants.toolkitTestName}::$v,toolkit")
+              .withScopeRequirement(Scope.Test)
+          )
+        else
+          List(
+            Positioned(positions, dep"$org::${Constants.toolkitName}::$v,toolkit")
+              .withEmptyRequirements
+          )
   val handler: DirectiveHandler[Toolkit] = DirectiveHandler.derive
-  def buildOptions(t: Positioned[String]): Either[BuildException, BuildOptions] =
-    Right {
-      BuildOptions(
-        classPathOptions = ClassPathOptions(
-          extraDependencies = ShadowingSeq.from(List(resolveDependency(t)))
-        )
-      )
+  private def buildOptionsWithScopeRequirement(
```

Comment:
What do you think about this approach? I tried to simplify this function

https://github.com/VirtusLab/scala-cli/commit/6183b76aae8371cb7367101cbd4346a1cd164616

Reply:
I don't think your approach will work... for the standard `//> using toolkit` directive, we need `toolkit` in the main scope and `toolkit-test` in the test scope, while your solution puts both in `main`.
try it out with my tests if it's unclear.

Reply:
Right, It fixed it and now it should work - https://github.com/lwronski/scala-cli/commit/eb3eb6c0e86b81c0fdd1cfc7968286def922570c